### PR TITLE
Karma Debug

### PR DESCRIPTION
**Pull request by trainerbill**
Karma debugging.  The tests pass in phantomjs if you use the --debug flag.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -82,3 +82,16 @@ Build checkout.js:
...
+```
+gulp test
+```
+
+| Flags  | Description |
+| ------------- | ------------- |
+| --clear | Clear Babel Loader and PhantomJS cache |
+| --debug | Debug mode.  PhantomJS, Karma, and CheckoutJS  |
+| --quick | Fastest testing.  Minimal output, no coverage |
+| --browsers | Choose Browser |
```
Comment (bluepnume) [inline README.md]: --browser

File: karma.conf.js [hunk 2]
```diff
@@ -2,203 +2,193 @@ let argv = require('yargs').argv;
...
 
+module.exports = function(config) {
 
-        // Continuous Integration mode
-        // if true, Karma captures browsers, runs the tests and exits
-        singleRun: true,
+    karmaConfig. logLevel = argv.debug ? config.LOG_DEBUG : config.LOG_WARN;
+    
+    if (!argv.quick) {
+        // addCoverage();
```
Comment (bluepnume) [inline karma.conf.js]: This should be enabled by default right?

File: karma.conf.js [hunk 3]
```diff
@@ -2,203 +2,193 @@ let argv = require('yargs').argv;
...
+function addPhantom() {
+    karmaConfig.phantomjsLauncher = {
+        exitOnResourceError: true
+    };
+    karmaConfig.customLaunchers.PhantomJS_custom = {
+        base: 'PhantomJS',
+        options: {
+            windowName: 'my-window',
+            settings: {
+                webSecurityEnabled: false
```
Comment (bluepnume) [inline karma.conf.js]: Can we leave web security enabled? Or do you see problems with it. Tests pass for me with this on, and I'd prefer to be closer to production settings.

File: gulpfile.babel.js [hunk 4]
```diff
@@ -51,16 +52,12 @@ gulp.task('lint-test', function() {
 
 gulp.task('karma', function (done) {
 
+    if(yargs.argv.clear) {
```
Comment (bluepnume) [inline gulpfile.babel.js]: Can we have `--clear-cache`? And also clear the babel cache maybe?

File: karma.conf.js [hunk 5]
```diff
@@ -2,203 +2,193 @@ let argv = require('yargs').argv;
...
-                        }
-                    },
-                    {
-                        test: /\.(html?|css|json)$/,
-                        loader: 'raw-loader'
-                    }
-                ]
+}
+
+function addPhantom() {
```
Comment (bluepnume) [inline karma.conf.js]: Can we not just put these in the config by default?

File: gulpfile.babel.js [hunk 6]
```diff
@@ -51,16 +52,12 @@ gulp.task('lint-test', function() {
...
+        del.sync(['node_modules/.cache/babel-loader', 'node_modules/.cache/phantomjs']);
+    }
+
     let server = new karma.Server({
         configFile: __dirname + '/karma.conf.js',
-        singleRun: !Boolean(yargs.argv['keep-browser-open']),
```
Comment (bluepnume) [inline gulpfile.babel.js]: Can we keep `keep-browser-open` too? Or maybe just `keep-open`

File: gulpfile.babel.js (hunk 4 above)
Comment (trainerbill) [inline gulpfile.babel.js]: it is clearing babel loader... is there another babel cache?

File: karma.conf.js (hunk 5 above)
Comment (trainerbill) [inline karma.conf.js]: makes it cleaner if you aren't using phantom.

File: gulpfile.babel.js (hunk 6 above)
Comment (trainerbill) [inline gulpfile.babel.js]: done

File: karma.conf.js (hunk 2 above)
Comment (trainerbill) [inline karma.conf.js]: Yeah.  missed it since tests never end on my machine.

File: gulpfile.babel.js (hunk 4 above)
Comment (bluepnume) [inline gulpfile.babel.js]: oh shoot, I can't read.

**Comment by bluepnume** [comment]
Looks good! Ready for merge? Sorry I took so long getting to reviewing your stuff, been a pretty hectic week so far.

**Pull request merged**